lib/options.py

```diff
@@ -205,6 +205,16 @@ def __init__(self):
                 2 x CPU count). -1 means everything running consistently
                 (single process). """)
 
+        parser.add_argument(
+                "-r", "--retries",
+                dest='retries_count',
+                default=env_int('TEST_RUN_RETRIES_COUNT', 3),
```

Comment:
IMO the default value should be 0 and we should explicitly set the retry count for CI runs. Rationale: when a developer writes a new test and runs test-run manually, he expects failures so he has to set the retry counter to 0.

Reply:
To be honest, I also thought about this in such a way. I like your idea and it is close to my feeling how should be right. But I had a small conversation about it with @Totktonada, maybe a month ago, and AFAIR he was sceptical about the default value == 0. AFAIR, @Totktonada was worry about the number of places where we need to do changes to enable auto-rerun. But it shouldn't be a hard task because at least in tarantool we have `environment` action where we can set the corresponding env variable `TEST_RUN_RETRIES_COUNT` and it automatically propagates through all our workflows because all of them use `environment` action. So I am for the default value == 0.

Reply:
We discussed it with Yaroslav. There are points for and against, but is seems that we should just solve the problem, which is more painful at the moment and look what will going on. IOW, I'm okay to set the default to "don't retry".

----

For the sake of completeness, points for the "retry 3 times" default:

* Infra: we don't need to proxy the variable to all environments in CI (to packpack, which in turn passes it to rpmbuild/debuild; to a docker container; to a virtual machine). It is implemented now, but may cost a time for support.
* Dev: if we'll apply the "no retry" default, running the full test suite will require more manual work to retry failed attempts (when irrelevant tests will fail). Since those attempts are around 5 minutes, context switching here is not a nice thing.

Points for the "no retry" default:

* Dev: no extra test-run specific knowledge is necessary to debug a test.